### PR TITLE
Extract minimal (orient-dependency free) entity api to nexus-commom

### PR DESCRIPTION
**Pull request by jdillon**
... and refactor blobstore configuration bits to use entity api and orient entity-adapter helpers.
Some tidy and name normalization as well.

Related to this effort: https://issues.sonatype.org/browse/NEXUS-6897

Related:

https://github.com/sonatype/nexus-pro/pull/497

### CODE REVIEW

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF78-3 :white_check_mark:

File: components/nexus-blobstore-api/src/main/java/org/sonatype/nexus/blobstore/api/BlobStoreMetrics.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
 package org.sonatype.nexus.blobstore.api;
 
 /**
- * Provides usage metrics for the blob store.
+ * Provides usage metrics for a blob-store.
```
Comment (kellyrob99) [inline components/nexus-blobstore-api/src/main/java/org/sonatype/nexus/blobstore/api/BlobStoreMetrics.java]: @link BlobStore

File: components/nexus-core/src/test/java/org/sonatype/nexus/internal/blobstore/BlobStoreConfigurationStoreImplTest.groovy [hunk 2]
```diff
@@ -79,7 +83,8 @@ class BlobStoreConfigurationStoreImplTest
       fail()
     }
     catch (ORecordDuplicatedException e) {
-      assert e.toString().contains('blobstore_name_idx')
+      // FIXME: This is fragile for refactoring
```
Comment (kellyrob99) [inline components/nexus-core/src/test/java/org/sonatype/nexus/internal/blobstore/BlobStoreConfigurationStoreImplTest.groovy]: should be sufficient now to just remove the index name check, as there's only one index that could be affected

**Comment by kellyrob99** [comment]
+1

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
+1

**Comment by mrprescott** [comment]
+1

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**